packages/rest-client/src/main/types/AuthOptions.ts

```diff
@@ -0,0 +1,16 @@
+import { User } from './User'
+
+/**
+ * Represents how we want to authenticate the call:
+ * - client: Are we using a system or user token?
+ * - user: For system calls, we only need username; for user calls, we only need token.
+ */
+export interface AuthOptions {
+  client: Token
```

Comment:
Think maybe `clientType: ClientType` or `tokenType: TokenType` ?

Reply:
Actioned in https://github.com/ministryofjustice/hmpps-typescript-lib/pull/54/commits/74d4cc02629099811212e0cdcb3a0f8f41c95e1f